.github/workflows/pr-tests.yml

```
@@ -0,0 +1,200 @@
+          pip install -r requirements-dev.txt -c constraints.txt
+
+      - name: Ruff Lint
+        run: ruff check . --output-format=github
+
+      - name: Black Check
+        run: black --check --diff .
+
+      - name: Determine changed files (PR)
+        if: github.event_name == 'pull_request'
+        id: changed
+        run: |
```

actionlint (1.7.8)

shellcheck reported issue in this script: SC2086:info:6:29: Double quote to prevent globbing and word splitting

(shellcheck)